rendermodules/point_match_optimization/pt_match_optimization.py

```diff
@@ -0,0 +1,583 @@
+        "port": 8080,
+        "owner": "gayathri",
+        "project": "MM2",
+        "client_scripts": "/allen/aibs/pipeline/image_processing/volume_assembly/render-jars/production/scripts"
+    },
+    "SIFT_options": {
+        "SIFTfdSize": [4, 6, 8],
+        "SIFTmaxScale": [0.82],
+        "SIFTminScale": [0.28],
+        "SIFTsteps": [4, 6, 8],
+        "renderScale": [0.1, 0.2],
+        "matchModelType": ["SIMILARITY", "RIGID", "AFFINE"]
+    },
+    "url_options":{
+        "normalizeForMatching": "true",
+        "renderWithFilter": "true",
+        "renderWithoutMask": "true"
+    },
+    "outputDirectory": "/allen/programs/celltypes/workgroups/em-connectomics/gayathrim/scratch/pmopts",
+    "tile_stack": "point_match_optimization_test",
+    "stack": "mm2_acquire_8bit_reimage_postVOXA_TEMCA2_rev1039",
+    "tilepair_file": "/allen/programs/celltypes/workgroups/em-connectomics/gayathrim/nc-em2/Janelia_Pipeline/scratch/fine/tilePairs/MM2/tile_pairs_mm2_acquire_8bit_reimage_postVOXA_TEMCA2_Rough_rev1039_v2_z_1023_to_1034_dist_2_p000.json",
+    "filter_tilepairs":False,
+    "no_tilepairs_to_test":5,
+    "max_tilepairs_with_matches":4  
+}
+
+ex = {
+    "render": {
+        "host": "http://em-131db",
+        "port": 8081,
+        "owner": "timf",
+        "project": "17797_1R",
+        "client_scripts": "/allen/aibs/pipeline/image_processing/volume_assembly/render-jars/production/scripts"
+    },
+    "SIFT_options": {
+        "SIFTfdSize": [4],
+        "SIFTmaxScale": [0.82],
+        "SIFTminScale": [0.28],
+        "SIFTsteps": [8],
+        "renderScale": [0.3],
+        "matchModelType": ["AFFINE"]
+    },
+    "url_options":{
+        "normalizeForMatching": "true",
+        "renderWithFilter": "true",
+        "renderWithoutMask": "true"
+    },
+    "outputDirectory": "/allen/programs/celltypes/workgroups/em-connectomics/gayathrim/scratch/ptmatch",
+    "tile_stack": "point_match_optimization_test",
+    "stack": "em_2d_montage_apply_mipmaps",
+    "tilepair_file": "/allen/programs/celltypes/workgroups/em-connectomics/gayathrim/scratch/17797_1R/tilepairs/temca4_tilepair_test.json",
+    "filter_tilepairs":False,
+    "no_tilepairs_to_test":4,
+    "max_tilepairs_with_matches":4  
+}
+
+
+def polys_overlap(ts, buffer=0):
+    ts1 = ts[0]
+    ts2 = ts[1]
+    p1 = Polygon(ts1.bbox_transformed(ndiv_inner=1))
+    p2 = Polygon(ts2.bbox_transformed(ndiv_inner=1))
+    return p1.intersects(p2)
+
+
+'''
+def get_canvas_url(render_args, stack, tile1, tile2, url_options):
+    base_data_url = '%s:%d/render-ws/v1'%(render_args['host'], render_args['port'])
+    tile_base_url = "%s/owner/%s/project/%s/stack/%s/tile"%(base_data_url,
+                                                            render_args['owner'],
+                                                            render_args['project'],
+                                                            stack)
+    url_suffix = "render-parameters"
```

Comment:
These are not all the urlOptions that are available.  I think either we should list them all here explicitly or have a wrapper function for converting them to query arguments in a standard fashion and accept them all.  Noteably, its missing handling of the excludeAllTransforms, excludeFirstTransformAndAllAfter, excludeTransformsAfterLast.

Reply:
These options are still not wired up in this part of the code.. I since we schematized them we should just have a input as a dictionary of query parameters arguments and append them all to the url, rather than listed them out explicitly, if someone passes an invalid query argument to this function it won't break anything, it just won't complain.  We could re-validate the dictionary with the schema in this function, but I think that's overkill.

Reply:
changes have been made

Reply:
removed it